Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -191,5 +207,12 @@ final class ConfigGenerator: ConfigGenerating {
                 }
             }
         }
+
+        if let deploymentTarget = target.deploymentTarget {
+            switch deploymentTarget {
+            case let .iOS(version, _): settings["IPHONEOS_DEPLOYMENT_TARGET"] = .string(version)
+            case let .macOS(version): settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
```

Comment:
Setting a version like this is very naive - we should definitely introduce some version validation. Wondering what would be the best place to do so. Maybe even in a public `Target` description? 

```
// Project.swift

Target(..., deploymentTarget: iOS(.v13_1, [.iphone, .ipad])
```

The disadvantage is that with every new os version, we will need to make the changes in the project.

Thoughts?

Reply:
You can model it like:

```swift
enum iOSVersion: CustomStringConvertible {
   case v13
   case v13_1
   case custom(String)
}
```
And then people could pass their own value if we are not listing recent versions.

However, even though this adds a bit of safety to ensure that they are introducing a valid version of iOS, I see two drawbacks that make me think that we should stay with a `String` as the value type:
- We need to keep the list up to date and therefore someone has to remember to update the enum when new versions of the runtimes are released.
- They can workaround the safety anytime by passing the string value directly.

If we keep in mind too that attribute is one of the first things that Xcode validates when you start compiling the app, they'll spot quickly if they set a value that is incorrect.

Reply:
Very well, so I'd leave only the string value then. 

> If we keep in mind too that attribute is one of the first things that Xcode validates when you start compiling the app, they'll spot quickly if they set a value that is incorrect.

Does it mean that we do not need to make any validation on our side if the string value is correct? I mean a user can potentially type something like: 

```
deploymentTarget: .macOS("this does not look like a correct version") 🙈
```

instead of:

```
deploymentTarget: .macOS("10.14.1") 👍
```

Reply:
We can add it, but I think the scenario that you are presenting is very unlikely. I think it's very obvious from the API that the expected string needs to be a version and not a sentence :trollface:.

Although you know that users can do anything...

Reply:
Cool, so what would be the best place to put such validation? `SettingsLinter`? 🤔

Reply:
Since the deployment target attribute belongs to the target, I'd rather place it in the `TargetLinter`. What do you think?

Reply:
Absolutely, makes sense 💪 I missed that there's `TargetLinter`.